storage/remote/write.go

```diff
@@ -177,13 +186,61 @@ func (rws *WriteStorage) Appender(_ context.Context) storage.Appender {
 	}
 }
 
+// timedCheckpointRecord records the Checkpoint Record every set time
+func (rws *WriteStorage) startTimedRecording() {
+	for {
+		<-time.After(time.Duration(saveTime) * time.Millisecond)
+		rws.writeCheckpointRecordJSON()
+	}
+}
+
+// writeCheckpointRecordJSON writes the segment record in a JSON file
+func (rws *WriteStorage) writeCheckpointRecordJSON() {
+	filepath := "data/CheckpointRecord.json"
+
+	rws.checkpoint.TimeRecorded = time.Now()
+	rws.checkpoint.Checkpoints = rws.getEndpointRecordData()
+
+	// Marshal json
+	data, err := json.MarshalIndent(rws.checkpoint, "", "")
```

Comment:
should the last parameter not be `"\t"`?

Reply:
oh it is. ill add it in